components/dash-core-components/src/utils/computeSliderMarkers.js

```diff
@@ -0,0 +1,135 @@
+import {pickBy} from 'ramda';
+
+/**
+ * Truncate marks if they are out of Slider interval
+ */
+const truncateMarks = (min, max, marks) =>
+    pickBy((k, mark) => mark >= min && mark <= max, marks);
+
+const truncateNumber = num =>
+    parseInt(num.toString().match(/^-?\d+(?:\.\d{0,0})?/)[0], 10);
+
+const decimalCount = d =>
+    String(d).split('.').length > 1 ? String(d).split('.')[1].length : 0;
+const alignIntValue = (v, d) =>
+    d < 10
+        ? v
+        : parseInt((truncateNumber(v / d) * d).toFixed(decimalCount(d)), 10);
+const alignDecimalValue = (v, d) =>
+    d < 10
+        ? parseFloat(v.toFixed(decimalCount(d)))
+        : parseFloat(((v / d).toFixed(0) * d).toFixed(decimalCount(d)));
+
+const alignValue = (v, d) =>
+    decimalCount(d) < 1 ? alignIntValue(v, d) : alignDecimalValue(v, d);
+
+const log = v => Math.floor(Math.log10(v));
+
+const getNearByStep = v =>
+    v < 10
+        ? [v]
+        : [
+              Math.pow(10, Math.floor(Math.log10(v))),
+              Math.pow(10, Math.ceil(Math.log10(v))) / 2,
+              alignValue(v, Math.pow(10, log(v))),
+              Math.pow(10, Math.ceil(Math.log10(v))),
+          ].sort((a, b) => Math.abs(a - v) - Math.abs(b - v));
+
+const estimateBestSteps = (minValue, maxValue, stepValue) => {
+    const desiredCountMin = 2 + (maxValue / stepValue <= 10 ? 3 : 3); // including start, end
+    const desiredCountMax = 2 + (maxValue / stepValue <= 10 ? 4 : 6);
+
+    const min = minValue / stepValue;
+    const max = maxValue / stepValue;
+
+    const rangeLength = max - min;
+
+    const leastMarksInterval = Math.max(
+        Math.round(rangeLength / (desiredCountMin - 1)),
+        1
+    );
+    const possibleValues = getNearByStep(leastMarksInterval);
+
+    const finalStep =
+        possibleValues.find(step => {
+            const expectedSteps = Math.ceil(rangeLength / step) + 1;
+            return (
+                expectedSteps >= desiredCountMin - 1 &&
+                expectedSteps <= desiredCountMax + 1
+            );
+        }) || possibleValues[0];
+    return [
+        alignValue(min, finalStep) * stepValue,
+        alignValue(finalStep * stepValue, stepValue),
+    ];
+};
+
+export const autoGenerateMarks = (min, max, step = 1) => {
```

Comment:
If user passes in `step`, then we should respect it. So I think you can remove ` = 1` here and add a `interval = step ? step : estimateBestSteps(min, max, step)`